### ISSUE TITLE
Xerces 2.12: problems with betterform <xf:bind/>

### ISSUE THREAD

**Issue by pherk** [eXist-db/exist#2007]
### What is the problem

the update to xerces 2.12 leads to NullPointerExecption if betterform is used.
the problem seems to be a call via <xf:bind id="" ref="" type="xs:int"/>.
Actually only the datatype xs:string is ok (xf:integer, xf:boolean tested with error result).

```
Caused by: java.lang.NullPointerException
946     at org.apache.xerces.impl.dv.xs.XSSimpleTypeDecl.getActualValue(Unknown Source) ~[xercesImpl-2.12.0.jar:?]
947     at org.apache.xerces.impl.dv.xs.XSSimpleTypeDecl.validate(Unknown Source) ~[xercesImpl-2.12.0.jar:?]
948     at de.betterform.xml.xforms.model.constraints.Validator.checkDatatype(Validator.java:303) ~[betterform-exist-5.1-SNAPSHOT-20160615.jar:?]
949     at de.betterform.xml.xforms.model.constraints.Validator.validate(Validator.java:257) ~[betterform-exist-5.1-SNAPSHOT-20160615.jar:?]
950     at de.betterform.xml.xforms.model.constraints.Validator.validate(Validator.java:213) ~[betterform-exist-5.1-SNAPSHOT-20160615.jar:?]
951     at de.betterform.xml.xforms.model.constraints.Validator.validate(Validator.java:190) ~[betterform-exist-5.1-SNAPSHOT-20160615.jar:?]
```

### What did you expect

I know it is a betterform problem not a eXist, but it is relevant for some eXist users.
betterform xf:bind should be usable with type validation

### Describe how to reproduce or add a test

as far I could reduce it.
some test app for betterform is needed (a little bit complex but the gurus knows ;-)
and the following example app function:

--
 see next comment
---

reverting to xerces 2.11 resolves the problem

### Context information
8 Jul 2018 11:20:08,957 [main] INFO  (JettyStart.java [run]:146) - Running with Java 1.8.0_171 [Oracle Corporation (Java HotSpot(TM) 64-Bit Server VM) in /usr/lib/jvm/java-8-oracle/jre]
08 Jul 2018 11:20:08,957 [main] INFO  (JettyStart.java [run]:153) - Running as user 'peter'
08 Jul 2018 11:20:08,957 [main] INFO  (JettyStart.java [run]:154) - [eXist Home : /media/peter/TrekStor/existdb-4.0]
08 Jul 2018 11:20:08,958 [main] INFO  (JettyStart.java [run]:155) - [eXist Version : 4.3.0]
08 Jul 2018 11:20:08,959 [main] INFO  (JettyStart.java [run]:156) - [eXist Build : 201807081118]
08 Jul 2018 11:20:08,959 [main] INFO  (JettyStart.java [run]:157) - [Git commit : dd335e5]
08 Jul 2018 11:20:08,959 [main] INFO  (JettyStart.java [run]:159) - [Operating System : Linux 4.4.0-130-generic amd64]
no relevant changes to conf.xml

**Comment by pherk** [comment]
`xquery version "3.1";

module namespace app="http://enahar.org/exist/apps/test";

declare namespace  xf="http://www.w3.org/2002/xforms";

declare function app:bind($node as node(), $model as map(*)) {
(
    <div style="display:none">
        <xf:model id="m-cps">
            <xf:instance xmlns="" id="iiter">
                <data >
                    <iter xmlns="http://hl7.org/fhir" value="true"/>
                </data>
            </xf:instance>

            <xf:bind ref="*:iter/@value" type="xs:boolean"/>
        </xf:model>
    </div>
,   <xf:group>
        <strong>Hello</strong>
        <div><br/>
            <p>it is {current-dateTime()}</p>
        </div>
    </xf:group>
)
};
`

**Comment by pherk** [comment]
---
xquery version "3.1";

module namespace app="http://enahar.org/exist/apps/test";

declare namespace  xf="http://www.w3.org/2002/xforms";

declare function app:bind($node as node(), $model as map(*)) {
(
    <div style="display:none">
        <xf:model id="m-cps">
            <xf:instance xmlns="" id="iiter">
                <data >
                    <iter xmlns="http://hl7.org/fhir" value="true"/>
                </data>
            </xf:instance>

            <xf:bind ref="*:iter/@value" type="xs:boolean"/>
        </xf:model>
    </div>
,   <xf:group>
        <strong>Hello</strong>
        <div><br/>
            <p>it is {current-dateTime()}</p>
        </div>
    </xf:group>
)
};

---

**Comment by pherk** [comment]
~~~
xquery version "3.1";

module namespace app="http://enahar.org/exist/apps/test";

declare namespace  xf="http://www.w3.org/2002/xforms";

declare function app:bind($node as node(), $model as map(*)) {
(
    <div style="display:none">
        <xf:model id="m-cps">
            <xf:instance xmlns="" id="iiter">
                <data >
                    <iter xmlns="http://hl7.org/fhir" value="true"/>
                </data>
            </xf:instance>
<!--
            <xf:bind id="biiter" ref="instance('iiter')/*:iiter" type="xs:string"/>
-->
            <xf:bind ref="*:iter/@value" type="xs:boolean"/>
        </xf:model>
    </div>
,   <xf:group>
        <strong>Hello</strong>
        <div><br/>
            <p>it is {current-dateTime()}</p>
        </div>
    </xf:group>
)
};

~~~

**Comment by dizzzz** [comment]
@JoernT can you have a look? xerces bug? or changed API semantics ?

**Comment by dizzzz** [comment]
@pherk is there a simple way for me to verify your code?  I could either replace xerces 2.12 with the version that does not support XSD1.1 or downgrade...

**Comment by pherk** [comment]
send a small test xar to @dizzzz per email 
-> localhost:8080/exist/apps/test/bind.html shows
   exception with xerces-2.12 and a page with 
   Hello it is 2018-07-10T19:04:56.643+02:00 with xerces-2.11

**Comment by duncdrum** [comment]
so merging into RC branch won't autoclose this @adam close or wait for it to make its way into master?

**Comment by adamretter** [comment]
It's fixed in 4.3.x
